pkg/sources/awssqs/awssqs.go

```diff
@@ -0,0 +1,214 @@
+	// kubeclientset is a standard kubernetes clientset
+	kubeclientset kubernetes.Interface
+	image         string
+	// namespace where the feeding is created
+	feedNamespace string
+	// serviceAccount that the container runs as. Launches Receive Adapter with the
+	// same Service Account
+	feedServiceAccountName string
+}
+
+func NewAWSSQSEventSource(kubeclientset kubernetes.Interface, feedNamespace string, feedServiceAccountName string, image string) sources.EventSource {
+	glog.Infof("Image: %q", image)
+	return &AWSSQSEventSource{kubeclientset: kubeclientset, feedNamespace: feedNamespace, feedServiceAccountName: feedServiceAccountName, image: image}
+}
+
+func (a *AWSSQSEventSource) StopFeed(trigger sources.EventTrigger, feedContext sources.FeedContext) error {
+	glog.Infof("Stopping AWS SQS Events feed with context %+v", feedContext)
+
+	deploymentName := feedContext.Context[deployment].(string)
+
+	err := a.deleteAWSSQSDeployment(deploymentName)
+	if err != nil {
+		glog.Warningf("Failed to delete deployment: %s", err)
+		return err
+	}
+	return nil
+}
+
+func (a *AWSSQSEventSource) StartFeed(trigger sources.EventTrigger, route string) (*sources.FeedContext, error) {
+	glog.Infof("creating awssqs feed context")
+	// create aws sqs deployment
+	awsToken := ""
+	awsId := trigger.Parameters["AWS_ACCESS_KEY_ID"].(string)
+	awsKey := trigger.Parameters["AWS_SECRET_ACCESS_KEY"].(string)
+	if token, ok := trigger.Parameters["AWS_SESSION_TOKEN"].(string); ok {
+		awsToken = token
+	}
+	if len(awsId) == 0 && len(awsKey) == 0 {
+		return nil, fmt.Errorf("missing AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY")
+	}
+	resource := strings.Split(trigger.Resource, "/")
+	if len(resource) != 2 {
+		return nil, fmt.Errorf("invalid resource: must be region/queue-name")
+	}
+	region := resource[0]
+	queueName := resource[1]
+
+	// create aws sqs client
+	sess := session.New(&aws.Config{
+		Region:      aws.String(region),
+		Credentials: credentials.NewStaticCredentials(awsId, awsKey, awsToken),
+	})
+	sqsClient := sqs.New(sess)
+	// list queue
+	param := &sqs.ListQueuesInput{
+		QueueNamePrefix: &queueName,
+	}
+	resp, err := sqsClient.ListQueues(param)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list queue %q: %v", queueName, err)
+	}
+	queueUrl := ""
+	for i := range resp.QueueUrls {
+		q := *resp.QueueUrls[i]
+		// if both prefix and suffix match, return it
+		if strings.HasSuffix(q, queueName) {
+			queueUrl = q
+			break
+		}
+	}
+	// if queue not found, create one
+	if len(queueUrl) == 0 {
+		param := &sqs.CreateQueueInput{
+			QueueName: &queueName,
+		}
+		glog.Infof("create queue %s", queueName)
+		resp, err := sqsClient.CreateQueue(param)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create queue %v", err)
+		}
+		queueUrl = *resp.QueueUrl
+	}
+
+	// create a random deployment name
+	uuid, err := uuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+	deploymentName := "awssqs-" + queueName + "-" + uuid.String()
```

Comment:
uuid is a very large number. Paul had a better method by using k8s generate name. @pmorie can you link to that api to get the short random name?